sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,577 @@
+
+    def get_vendor(self, port):
+        """
+        This API specifically returns the vendor name of the Y cable for a specfic port.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+        Returns:
+            a string, with vendor name
+        """
+
+        raise NotImplementedError
+
+    def get_part_number(self, port):
+        """
+        This API specifically returns the part number of the Y cable for a specfic port.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+        Returns:
+            a string, with part number
+        """
+
+        raise NotImplementedError
+
+    def get_switch_count(self, port, count_type):
+        """
+        This API specifically returns the switch count to change the Active TOR which has
+        been done manually by the user.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            count_type:
+                 a string, for getting the count type
+                          "manual" -> manual switch count
+                          "auto" -> automatic switch count
+        Returns:
+            an integer, the number of times manually the Y-cable has been switched
+        """
+
+        raise NotImplementedError
+
+    def get_target_cursor_values(self, port, lane, target):
+        """
+        This API specifically returns the cursor equalization parameters for a target(NIC, TOR1, TOR2).
+        This includes pre one, pre two , main, post one, post two cursor values
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            lane:
+                 an Integer, the lane on which to collect the cursor values
+                             1 -> lane 1,
+                             2 -> lane 2
+                             3 -> lane 3
+                             4 -> lane 4
+            target:
+                 an Integer, the actual target to get the cursor values on
+                             TARGET_NIC -> NIC,
+                             TARGET_TOR1-> TOR1,
+                             TARGET_TOR2 -> TOR2
+        Returns:
+            a list, with  pre one, pre two , main, post one, post two cursor values in the order
+        """
+
+        raise NotImplementedError
+
+    def get_firmware_version(self, port, target):
+        """ This routine should return the active, inactive and next (committed)
+        firmware running on the target. Each of the version values in this context
+        could be a string with a major and minor number and a build value.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+            target:
+                 an Integer, the actual target to get the cursor values on
```

Comment:
Maybe rather than explicitly stating "An Integer", we simply state "One of the following predefined constants:". This would also allow us to change the underlying type transparently if desired.

Reply:
fixed